### DISCUSSION TITLE
what happens with overly-long captions?

### DISCUSSION THREAD

**Discussion by caniyabanci76** [kohya-ss/sd-scripts#946] [Q&A]
during training, what happens with overly-long captions? if they tokenize to over 77 tokens, what happens to the remainder?  is it truncated?  treated as 2 separate chunks?

**Comment by araleza** [comment]
I think it will work fine as long as you've set the max token length to 150 instead of 75.  You can do that in the Advanced tab of the webui gui for kohya, or pass '--max_token_length=150' if you're using sd-scripts directly from the command line.

That option doesn't seem to work for the samples that are generated, but that won't affect the training.  There's someone asking for it to work for the samples here:

https://github.com/kohya-ss/sd-scripts/issues/951

**Reply by caniyabanci76** [reply]
is it similar to what Automatic1111 WebUI does in that it processes each 75 chunk separately, and then concatenates the result?

**Comment by Serallan** [comment]
A bit of a late reply, but I've been wondering the same. Take all of this with a truckload of salt because I'm not a coder, just a hobbyist and could be interpreting things wrong.

A look into the code shows it's truncated:

```
# stop if the text is too long (longer than truncation limit)
if len(text_token) > max_length:
    truncated = True
```
Testing this and the associated functions shows that tokens over the limit are discarded. I pulled that one from `parse_prompt_attention()` in `gen_img_diffusers.py`.The same functions also show up in another files and I'm not familiar enough with the code to tell if this is used in training or just sampling.
